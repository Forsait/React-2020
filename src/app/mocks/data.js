export const film = {
  budget: 0,
  genres: ['Action', 'Adventure', 'Drama', 'Science Fiction', 'Thriller'],
  id: 351286,
  overview: 'A volcanic eruption threatens',
  poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
  release_date: '2018-06-01',
  revenue: 0,
  runtime: null,
  tagline: 'Life finds a way',
  title: 'Jurassic World: Fallen Kingdom',
  vote_average: 0,
  vote_count: 28,
};

export const moviesData = {
  data: new Array(6).fill(film),
  limit: 0,
  offset: 0,
  total: 0,
};

export const radioOptions = {
  title: 'Radio title',
  list: [
    {
      key: 'key 1',
      title: 'title 1',
    },
    {
      key: 'key 2',
      title: 'title 2',
    },
  ],
};
