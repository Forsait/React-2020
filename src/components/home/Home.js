import React from 'react';

import Search from '../search/Search';
import MovieList from '../movie/MovieList';
import Footer from '../common/Footer';

const startSearchOptions = {
  searchBy: 'title',
  sortBy:'release_date',
}

export default class Home extends React.Component {

  constructor(props){
    super(props);
    this.searchFormSubmit = this.searchFormSubmit.bind(this);

    this.state = {
      result: {
        data: [],
        limit: 0,
        offset: 0,
        total: 0
      },
      totalResult: null
    };

    this.offset = 0;
    this.limit = 6;
  }

  searchFormSubmit(opt, skipUpdateTotal = false) {
    const url = `/api/movies?offset=${this.offset}&limit=${this.limit}&sortBy=${opt.sortBy}&search=${opt.query || ''}&searchBy=${opt.searchBy}&sortOrder=desc`
    fetch(url)
      .then(res => res.json())
      .then((res) => {
        console.log(res)
        this.setState({
          result: res,
          totalResult: skipUpdateTotal ? this.state.totalResult : res.total
        });
      })
  }

  componentDidMount() {
    this.searchFormSubmit(startSearchOptions, true);
  }

  render() {
    return (
      <>
        <Search resultTotal={this.state.totalResult} submit={this.searchFormSubmit}/>
        <MovieList movieArr={this.state.result.data}/>
        <Footer />
      </>
    )
  }
}