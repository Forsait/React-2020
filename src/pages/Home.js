import React from 'react';

import Search from '../containers/search/Search';
import MovieList from '../containers/movie/MovieList';
import Footer from '../components/common/Footer';

import { getMovieList } from '../services/movie.service';

const startSearchOptions = {
  searchBy: 'title',
  sortBy:'release_date',
}

export default class Home extends React.Component {

  constructor(props){
    super(props);
    this.searchFormSubmit = this.searchFormSubmit.bind(this);

    this.state = {
      result: {
        data: [],
        limit: 0,
        offset: 0,
        total: 0
      },
      totalResult: null
    };

    this.offset = 0;
    this.limit = 6;
  }

  searchFormSubmit(opt, skipUpdateTotal = false) {
    getMovieList(opt)
      .then(res => res.json())
      .then((res) => {
        console.log(res)
        this.setState({
          result: res,
          totalResult: skipUpdateTotal ? this.state.totalResult : res.total
        });
      })
  }

  componentDidMount() {
    this.searchFormSubmit(startSearchOptions, true);
  }

  render() {
    return (
      <>
        <Search resultTotal={this.state.totalResult} submit={this.searchFormSubmit}/>
        <MovieList movieArr={this.state.result.data}/>
        <Footer />
      </>
    )
  }
}