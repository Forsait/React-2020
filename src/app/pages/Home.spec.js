import React from 'react';
import { shallow } from 'enzyme';
import { act } from "react-dom/test-utils";

import { Home } from './Home';
import { film } from '../mocks/data';

describe('Home component', () => {

  it('should be render correctly', async () => {
    const movies = new Array(6).fill(film);
    const com = shallow(<Home movies={movies} />);
    expect(com).toMatchSnapshot()
  });

});
