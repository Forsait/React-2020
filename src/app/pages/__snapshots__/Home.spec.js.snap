// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should be render correctly 1`] = `
<Fragment>
  <Search
    resultTotal={null}
    submit={[Function]}
  />
  <MovieList
    movieArr={Array []}
  />
  <Footer />
</Fragment>
`;
